### DISCUSSION TITLE
Can the MONOSPACE text formatting be exposed to Python Dbus for signal-cli?

### DISCUSSION THREAD

**Discussion by hondogitsune** [AsamK/signal-cli#1421] [Q&A] [answered]
I'd like to be able to send signal-cli messages as pretty monospaced text. The formatting we know and love from both desktop and phone apps. Can we use it in signal-cli utilizing Python code over dbus?

I probably don't need to be able to select lengths and starting points. General support to make an entire message monospace for readability of sensor data would help a lot!

**Comment by AsamK** [comment, marked as answer]
Sending formatted text works with the jsonrpc mode, but not via dbus yet. See also #1271

**Comment by hondogitsune** [comment]
Thanks but the documentation is not really working for me. I see how it works in CLI mode, but I cannot find how it has to be called/named in JSON formatting.

**Reply by AsamK** [reply]
The format corresponds to the CLI commands, see https://github.com/AsamK/signal-cli/blob/master/man/signal-cli-jsonrpc.5.adoc#commands

For a single text style it would look like this:
`{"jsonrpc":"2.0","id":"id","method":"send","params":{"account":"+XXX","recipients":["+YYY"],"message":"HELLOs WW 1234IiP","textStyle":"0:17:MONOSPACE"}}`

Or like this for multiple styles:
`{"jsonrpc":"2.0","id":"id","method":"send","params":{"account":"+XXX","recipients":["+YYY"],"message":"HELLOs WW 1234IiP","textStyles":["0:17:MONOSPACE","4:10:BOLD"]}}`

**Reply by hondogitsune** [reply]
Thank you for your patience! I was able to implement it. I am still receiving over DBus but sending over JSON-RPC. You are an amazing developer for allowing parallel access to different methods for posting and receiving!

I am very happy with the results:
![grafik](https://github.com/AsamK/signal-cli/assets/75497157/42a068bf-34f0-428b-b8a2-453738ed0137)
